Core/src/org/sleuthkit/autopsy/geolocation/datamodel/WaypointBuilder.java

```diff
@@ -480,9 +481,10 @@ public void process(ResultSet rs) {
                             int artifact_type_id = rs.getInt("artifact_type_id"); //NON-NLS
                             long artifact_id = rs.getLong("artifact_id"); //NON-NLS
 
-                            BlackboardArtifact.ARTIFACT_TYPE type = BlackboardArtifact.ARTIFACT_TYPE.fromID(artifact_type_id);
-
-                            waypoints.addAll(getWaypointForArtifact(skCase.getBlackboardArtifact(artifact_id), type));
+                            ARTIFACT_TYPE type = ARTIFACT_TYPE.fromID(artifact_type_id);
```

Comment:
I am a little torn about whether to do this or include the types in the query.  Was there a reason that you didn't look at extending the query?  It might be too messy.  That figuring out the query was a pain.

Reply:
This just seemed like the right place to do type filtering. I don't really have a specific reason other than this seeming to be a good way to do it to me.

Reply:
So you didn't look into adjusting the query at all?

Reply:
I did, but I just settled on this. It sounds like you prefer I do the logic in the query?